genesyscloud/architect_datatable_row/resource_genesyscloud_architect_datatable_row_proxy.go

```diff
@@ -113,13 +122,29 @@ func getAllArchitectDatatableFn(ctx context.Context, p *architectDatatableRowPro
 
 		for _, table := range *tables.Entities {
 			totalRecords = append(totalRecords, table)
+			rc.SetCache(p.dataTableCache, *table.Id, *ConvertDatatable(table))
 		}
 	}
 
 	return &totalRecords, nil
 }
 
+func ConvertDatatable(master platformclientv2.Datatable) *Datatable {
+	var datatable Datatable
+	err := mapstructure.Decode(master, &datatable)
+	if err != nil {
+		return nil
+	}
```

Comment:
Are we losing some info here by not handling the error? I think we should at least log the error, even if it shouldn't cause the process to fail

Reply:
edgecase where platformclientv2.Datatable cannot be converted to our custom DataTable. I am returning null in this scenario and the cache miss happens during getCache and the export proceeds to do an API call. Will add a log statement here.